packages/charts/react-charting/src/components/DeclarativeChart/helpers.ts

```diff
@@ -0,0 +1,280 @@
+
+      svgToPng(s, {
+        width: opts.width || svg.width,
+        height: opts.height || svg.height,
+        scale: opts.scale,
+      })
+        .then(resolve)
+        .catch(reject);
+    } catch (err) {
+      return reject(err);
+    }
+  });
+}
+
+function toSVG(chartContainer: HTMLElement, background: string) {
+  const svg = chartContainer.querySelector<SVGSVGElement>('svg');
+  if (!svg) {
+    throw new Error('SVG not found');
+  }
+
+  const { width: svgWidth, height: svgHeight } = svg.getBoundingClientRect();
+  const classNames = new Set<string>();
+  const legendGroup = cloneLegendsToSVG(chartContainer, svgWidth, svgHeight, classNames);
+  const w1 = Math.max(svgWidth, legendGroup.width);
+  const h1 = svgHeight + legendGroup.height;
+  const clonedSvg = d3Select(svg.cloneNode(true) as SVGSVGElement)
+    .attr('width', null)
+    .attr('height', null)
+    .attr('viewBox', null);
+
+  if (legendGroup.node) {
+    clonedSvg.append(() => legendGroup.node);
+  }
+  clonedSvg
+    .insert('rect', ':first-child')
+    .attr('x', 0)
+    .attr('y', 0)
+    .attr('width', w1)
+    .attr('height', h1)
+    .attr('fill', background);
+
+  const svgElements = svg.getElementsByTagName('*');
+  const styleSheets = document.styleSheets;
+  const styleRules: string[] = [];
+
+  for (let i = svgElements.length - 1; i--; ) {
+    svgElements[i].classList.forEach(className => {
+      classNames.add(`.${className}`);
+    });
+  }
+
+  for (let i = 0; i < styleSheets.length; i++) {
+    const rules = styleSheets[i].cssRules;
+    for (let j = 0; j < rules.length; j++) {
+      if (rules[j].constructor.name === 'CSSStyleRule') {
+        const selectorText = (rules[j] as CSSStyleRule).selectorText;
+        const hasClassName = selectorText.split(' ').some(word => classNames.has(word));
+
+        if (hasClassName) {
+          styleRules.push(rules[j].cssText);
+        }
+      }
+    }
+  }
+
+  const xmlDocument = new DOMParser().parseFromString('<svg></svg>', 'image/svg+xml');
+  const styleNode = xmlDocument.createCDATASection(styleRules.join(' '));
+  clonedSvg.insert('defs', ':first-child').append('style').attr('type', 'text/css').node()!.appendChild(styleNode);
+
+  clonedSvg.attr('width', w1).attr('height', h1).attr('viewBox', `0 0 ${w1} ${h1}`);
+
+  return {
+    node: clonedSvg.node()!,
+    width: w1,
+    height: h1,
+  };
+}
+
+function cloneLegendsToSVG(chartContainer: HTMLElement, svgWidth: number, svgHeight: number, classNames: Set<string>) {
+  const legendButtons = chartContainer.querySelectorAll<HTMLElement>(`
+    button[class^="legend-"],
+    [class^="legendContainer-"] div[class^="overflowIndicationTextStyle-"],
+    [class^="legendsContainer-"] div[class^="overflowIndicationTextStyle-"]
+  `);
+  if (legendButtons.length === 0) {
+    return {
+      node: null,
+      width: 0,
+      height: 0,
+    };
+  }
+
+  const legendGroup = d3Create<SVGGElement>('svg:g');
```

Comment:
>legendGroup [](http://example.com/codeflow?start=8&length=11)

does the legend include overflow legends also

Reply:
Not with the current implementation, as overflowed legends are rendered in a hover card (callout) outside the chart DOM tree.